src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -170,11 +171,17 @@ private static FileStream CreateSharedBackingObject(Interop.Sys.MemoryMappedProt
                 ?? CreateSharedBackingObjectUsingFile(protections, capacity, inheritability);
         }
 
-        private static FileStream? CreateSharedBackingObjectUsingMemory(
+        private static unsafe FileStream? CreateSharedBackingObjectUsingMemory(
            Interop.Sys.MemoryMappedProtections protections, long capacity, HandleInheritability inheritability)
         {
-            // The POSIX shared memory object name must begin with '/'.  After that we just want something short and unique.
-            string mapName = string.Create(null, stackalloc char[128], $"/corefx_map_{Guid.NewGuid():N}");
+            const int MaxSharedMemoryObjectNameLength = 32; // SHM_NAME_MAX on OSX ARM64, on other systems it's equal PATH_MAX (250)
+            // The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
+            Span<char> characters = stackalloc char[MaxSharedMemoryObjectNameLength];
+            char format = 'N';
+            bool success = Guid.NewGuid().TryFormat(characters, out int charsWritten, new ReadOnlySpan<char>(&format, 1));
```

Comment:
`bool success = Guid.NewGuid().TryFormat(characters, out int charsWritten, "N".AsSpan());` avoids unsafe code?

Reply:
To be honest I was sure that it would allocate more memory, but seems not:

```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using System;

namespace MapName
{
    internal class Program
    {
        static unsafe void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);
    }

    public class Strings
    {
        [Benchmark(Baseline = true)]
        public string Current() => string.Create(null, stackalloc char[128], $"/corefx_map_{Guid.NewGuid():N}");

        [Benchmark]
        public unsafe string Mutable()
        {
            string simple = Guid.NewGuid().ToString("N");
            fixed (char* first = simple)
            {
                first[0] = '/';
            }
            return simple;
        }

        [Benchmark]
        public unsafe string Fix()
        {
            Span<char> characters = stackalloc char[32];
            char format = 'N';
            Guid.NewGuid().TryFormat(characters, out int charsWritten, new ReadOnlySpan<char>(&format, 1));
            characters[0] = '/';
            return new string(characters);
        }

        [Benchmark]
        public string Dan()
        {
            Span<char> characters = stackalloc char[32];
            Guid.NewGuid().TryFormat(characters, out int charsWritten, "N".ToString());
            characters[0] = '/';
            return new string(characters);
        }
    }
}
```

|  Method |     Mean |    Error |  StdDev | Ratio |  Gen 0 | Allocated |
|-------- |---------:|---------:|--------:|------:|-------:|----------:|
| Current | 192.2 ns | 23.32 ns | 1.28 ns |  1.00 | 0.0134 |     112 B |
| Mutable | 157.2 ns | 16.23 ns | 0.89 ns |  0.82 | 0.0105 |      88 B |
|     Fix | 165.5 ns | 27.24 ns | 1.49 ns |  0.86 | 0.0105 |      88 B |
|     Dan | 167.4 ns | 17.69 ns | 0.97 ns |  0.87 | 0.0105 |      88 B |

Reply:
Keep in mind that stackalloc's have more cost the larger the amount of space unless you've specified SkipLocalsInit, which the benchmark above doesn't have but all of our core assemblies do.  That's why "Current" in your chart shows up as so much larger than the others (that, and it's generating a longer string than the rest because it still has the longer prefix).

```C#
using System;
using System.Runtime.CompilerServices;
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

[module: SkipLocalsInit]

public class Program
{
    public static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);

    [Benchmark(Baseline = true)]
    public string Current() => string.Create(null, stackalloc char[128], $"/{Guid.NewGuid():N}"); // 33 char string, so not 100% comparable to the rest

    [Benchmark]
    public unsafe string Mutable()
    {
        string simple = Guid.NewGuid().ToString("N");
        fixed (char* first = simple)
        {
            first[0] = '/';
        }
        return simple;
    }

    [Benchmark]
    public unsafe string Fix()
    {
        Span<char> characters = stackalloc char[32];
        char format = 'N';
        Guid.NewGuid().TryFormat(characters, out int charsWritten, new ReadOnlySpan<char>(&format, 1));
        characters[0] = '/';
        return new string(characters);
    }

    [Benchmark]
    public string Dan()
    {
        Span<char> characters = stackalloc char[32];
        Guid.NewGuid().TryFormat(characters, out int charsWritten, "N");
        characters[0] = '/';
        return new string(characters);
    }

    [Benchmark]
    public string Steve()
    {
        return string.Create(32, 0, (span, state) =>
        {
            Guid.NewGuid().TryFormat(span, out int charsWritten, "N");
            span[0] = '/';
        });
    }
}
```

|  Method |     Mean |   Error |  StdDev | Ratio |
|-------- |---------:|--------:|--------:|------:|
| Current | 192.8 ns | 2.17 ns | 1.92 ns |  1.00 |
| Mutable | 186.9 ns | 1.12 ns | 0.94 ns |  0.97 |
|     Fix | 191.9 ns | 0.66 ns | 0.62 ns |  1.00 |
|     Dan | 190.7 ns | 1.50 ns | 1.33 ns |  0.99 |
|   Steve | 188.4 ns | 1.77 ns | 1.65 ns |  0.98 |

Reply:
(Though the cost of this operation is almost certainly dwarfed entirely by everything else that needs to happen as part of creating the MMF.)

Reply:
This conversation reminds me of https://meta.wikimedia.org/wiki/Cunningham%27s_Law 

> the best way to get the right answer on the internet is not to ask a question; it's to post the wrong answer.